packages/api/src/providers/fitbit.ts

```diff
@@ -373,7 +387,82 @@ export class Fitbit extends Provider implements OAuth2 {
     );
   }
 
-  // Creates a subscription for all or one specific collectionType
+  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
+  async revokeTokenFromExistingUser(
+    userId: string,
+    fitbitUserId: string,
+    activeSubscriptions: string[],
+    accessToken: string
+  ) {
+    try {
+      if (activeSubscriptions.length > 0) {
+        const connectedUsers = await getConnectedUsersByTokenOrFail(
+          ProviderSource.fitbit,
+          fitbitUserId
+        );
+        await Promise.all(
+          Object.values(connectedUsers).map(async user => {
+            if (user.dataValues.id !== userId) {
+              await this.oauth.revokeLocal(user);
+              await sendProviderDisconnected(user, [ProviderSource.fitbit]);
```

Comment:
this won't work, see thread [here](https://metriport.slack.com/archives/C04CXLSAF7V/p1690741639498519) - fitbit will still be in the provider map after calling `revokeLocal()`, which will lead to the behavior in #737.

also, we don't need to `await` the webhook, that can just be done async